### ISSUE TITLE
Investigate relabeling of items in PhET Menus

### ISSUE THREAD

**Issue by terracoda** [phetsims/a11y-research#12]
Based on Dec 5 meeting with @emily-phet and @jessegreenberg regarding the accessibility of the Sim Navbar, it was suggested that several items be made into buttons and be relabeled with clearer text. We basically have a list of buttons (items that do things) and links (items that go places).

**Current Menu Sample**
- Options... (Opens options dialog with checkboxes)
- PhET Website... (Link)
- Report a Problem... (Link)
- Screenshot (Takes a screenshot)
- Full Screen (Goes into full screen mode)
- About… (Opens dialog with details about this Sim)

**Draft Order, Labels, and Text to Discuss**
- Options (Opens options dialog with checkboxes) [Button]
  - View options for this sim.
- Take a Screenshot [Button]
  - Take and save a screenshot of the sim.
- Full Screen [Button]
  -  Use the sim in full screen mode.
- About this Sim [Button]
  - Design details about this sim.
- Report a Problem [Link]
  - Please let us know if you find a problem.
- PhET Website [Link]

**Questions**
- What do people think of the labels and help text above?
- Do we need help text?
- Do we need headings to label the menubar, and or the PhET Menu?

**Comment by terracoda** [comment]
This issue is related to #11.

**Comment by terracoda** [comment]
List of items can be viewed as tools and links:
**Sim Tools**
    Options (Opens options dialog with checkboxes) [Button]
        View options for this sim.
    Take a Screenshot [Button]
        Take and save a screenshot of the sim.
    Full Screen [Button]
        Use the sim in full screen mode.
    About this Sim [Button]
        Design details about this sim.
**Links**
    Report a Problem [Link]
        Please let us know if you find a problem.
    PhET Website [Link]
![sketch-phetmenu](https://cloud.githubusercontent.com/assets/1902116/20939052/e785a5b8-bbc7-11e6-9b7f-894f462cb03a.JPG)

**Comment by terracoda** [comment]
**Testing MENU element**

- FireFox: 
  - Menu type context with li’s or menuitems do not render at all. The entire contents of menuitem are essentially not there.
  - With type=”context” attribute removed, the li’s and menuitems’s render, but there are no list or group semantics read out.
  - Label attribute on menu is not read out
  - Number of li’s or menuitems are not read out
  - Tabindex=”0” is needed to make the menu (or div) focusable
  - I can’t seem to get aria-activedescendent to work
  - With VO on, the items are all read out back-to-back as a group when there is a tabindex=”0”
  - I can tab through items with or without VO and arrow and tab through when VO is on.
  - With role toolbar, I can arrow through the toolbar’s direct children which is nice.
  - Number of items in the menu only read out when a ul is used within a menu, not with menu alone. 
  - The are role “toolbar” is recognized by VO.
- **Summary:** Semantics are best when a ul is inside the menu element (or div element). Menu and div behave the same.
- Safari: 
  - Renders items with or without type=”context”. 
  - Safari behaves similarly to FF except in most keyboard navigation, except when the menu element is used as a parent to the list items. In this case, VO reads bullets and list items as separate things (annoying)
- **Summary:** Safari behaves similarly to FF in most keyboard navigation, except with the menu element. In this case, Safari rads both the bullet and the list item as separate things.

**Grand Summary:** 
I haven’t tried Chrome or any other screen reader, but it seems that the menu element does not add any extra semantics in FF and Safari. We need a ul element to have the list of items counted by the AT.

**Comment by terracoda** [comment]
@jessegreenberg, @jhung, @jobara would you guys like to chime in on this markup for the bar at the bottom of the sim. I left the sim title (H1) off as that is up at the top of the PDOM.

Note that aria-activedescendent attribute is not working. I think I need some JS for that?

The code over here:
[phet-templates](https://github.com/terracoda/phet-templates/blob/master/a11y-menu-template.html)

**Comment by jessegreenberg** [comment]
> Note that aria-activedescendent attribute is not working. I think I need some JS for that?

Yes, some script could update the aria-activedescendent id.  But is it necessary in this case?  Since each list item/button will be in the focus order, the AT will announce when it receives focus automatically.

For the menu, why did you choose the ```toolbar```  role instead of ```menubar```?

For https://github.com/terracoda/phet-templates/blob/master/a11y-menu-template.html#L38-L49, what do you think about using a an [ARIA menu button](https://www.w3.org/TR/wai-aria-practices-1.1/#menubutton).  I took it for a test drive in #13.

**Comment by jessegreenberg** [comment]
@terracoda thanks for testing the menu in https://github.com/phetsims/a11y-research/issues/12#issuecomment-265595630!

Have you tried the aria menu role in Safari with VO?

**Comment by jobara** [comment]
@jessegreenberg, in general I think it looks pretty good. The only comments I have right now are:

- Should the menulist have a `role="menu"` and then each item a `role="menuitem"`?
- I wonder if some of the `<div>` elements could/should be changed to `<section>` elements.

**Comment by terracoda** [comment]
@jessegreenberg, @jobara  I was just focusing on the HTML `menu`/`menuitem` elements. 

I will now test the `menu` and `menuitem` as aria roles using the list structure since the elements themselves provide no useful semantics.

I agree for aria roles that the `menubar` role is better than `toolbar`. 

@jobara as for using the more semantic `section` I was also thinking that would be good, but generally sections should contain a heading and I'm not sure we need a heading for each section here. We certainly shouldn't need a heading for the hotkeys button, for example.

I added some questions to the initial post in the issue. I've mixed up the 2 issues #11 and #12. Originally, I thought I would work out the code in #11, and the content in #12, but they are tightly related :-)

**Comment by emily-phet** [comment]
@terracoda I like the reorganization in your drawing above, though I'm not sure about your headings. Perhaps we can implement without the headings, and just have a horizontal line seperating the links from the buttons? I'm just a little concerned that the headings may be more confusing than helpful for those visually accessing the menu.

**Comment by terracoda** [comment]
@emily-phet, I actually missed one item (Check for Updates). I agree, too. I don't think I like the headings.
We could use icons to differentiate the tool-like things from the modal dialog with content in them.

Also wondering if "Report a Problem" can be shortened to "Report Problem." 

 How's this one and this order? I moved "Report a Problem" up.
![sketch_phetmenu](https://cloud.githubusercontent.com/assets/1902116/21020813/82442f3c-bd50-11e6-9cef-159f3291f568.jpg)

**Comment by terracoda** [comment]
@emily-phet, I was also thinking that instead of heading, non-visual help text could be "Tools and links for this sim." This text would be associated with the "PhET Menu" button, kind of like this:

- Hot Keys and Help (button)
  - Tips on how to use this sim with a keyboard.
- PhET Menu (button for unhiding the menu)
  - Tools and links for this sim.

**Comment by terracoda** [comment]
@emily-phet, notice that if read out together, they sound nice.

**Comment by terracoda** [comment]
@emily-phet, also don't think we need help text for every item.

- Hot Keys and Help: Tips on how to use this sim with a keyboard.
- PhET Menu: Tools and links for this sim.

Then the actual menu
- View Options
- Take a Screenshot
- Full Screen
- Report a Problem
- About this Sim

Visual separation here
- PhET Website

**Comment by terracoda** [comment]
@jessegreenberg and @jobara, I guess the code discussion can continue over in #13, I didn't realize @jessegreenberg was also working on this :-) 

It was a useful exercise to check the HTML elements since one should use HTML if it natively does what ARIA is supposed to add on.

**Comment by jessegreenberg** [comment]
@terracoda that looks good, I like the possibility of icons.  Why did you move report a problem up?  I thought one of the confusing parts of the menu was that the buttons and links were not collected together.

**Comment by jhung** [comment]
@jessegreenberg, @terracoda: @jobara and I just did a quick informal test of the menu with a colleague and here's what we found:

* Some dialogs are inconsistent in the use of the X close button. Options has it, the others do not.
* Check for Updates is unclear what it does. Sounds like it updates the sim.
* Check for Updates and About PhET seem like something that should go into "About" (actually, About has this information already).
* Not clear what Projector Mode actually does.
* The mixing of dialogs, links, and actions within a single menu didn't appear to cause any confusion.
* The Options dialog can be a bit strange. In the sims that have the "Option" item, there is only 1 option, so why have a dialog for just 1 option? 
* In the Molecule Shapes sim, there's an Option panel on the sim itself, as well as in the menu. This can cause confusion.
* Aside: the Molecule Shapes sim, there are 2 options in the Options dialog... which is a good use of that menu item I suppose.

**Comment by terracoda** [comment]
@jessegreenberg, I moved the "Report a Problem" up to group things by relevance to the sim rather than by function (i.e., button or link). The "Report a Problem" item is relevant to the sim. The last item in the sim relevant-section is the About.

The only item that really doesn't have anything to do with the sim is the link to the PhET website, so that was moved into its own section on the menu. 

Make sense?

**Comment by jessegreenberg** [comment]
That makes total sense, thanks for clarifying @terracoda.

**Comment by jessegreenberg** [comment]
@jhung that list is very helpful, thank you! It lists several of the confusing aspects of the current menu.  I think all of these points should be made into issues and considered.  To respond to a few of these issues:

> Some dialogs are inconsistent in the use of the X close button. Options has it, the others do not.

Very true, our dialogs are quite inconsistent throughout the project.  Some work is being done now to normalize our dialogs.

> Check for Updates is unclear what it does. Sounds like it updates the sim.

Yes, that is what it does.  Teachers requested that sims have this feature so that downloaded versions can check for updated versions on the website.  This is useful when internet may not be available during class hours.

> Not clear what Projector Mode actually does.

This is another feature for teachers.  When using a projector in a classroom, colors can look washed out so it changes the color profile so that sims an be easily seen on a projector.

> The Options dialog can be a bit strange. In the sims that have the "Option" item, there is only 1 option, so why have a dialog for just 1 option?

We wanted a general place for options for all sims.  In some sims there is only one option, but others have more than one, and this is where new global sim options could be placed in the future.

**Comment by terracoda** [comment]
@jhung thanks for the list about the individual items. That is indeed very helpful :-) It's all related of course to making the labels clear, but maybe the design changes for the actual modals we should probably put in a separate issue. I agree that they should be consistent and each contain a close button.

As for clarity of labels do you think that the labels could be made more clear with help text like the text I suggested at the beginning of this issue?

Or do you think that a general help text for the menu alone would help prepare users for what is in the menu. For example, "Tools and links for this sim" as I suggested above? 

I was hoping that small tweaks to the labels and perhaps the addition of icons could render the individual button help text irrelevant.

@jessegreenberg, thanks for the clarification on each @jhung's items.

**Comment by emily-phet** [comment]
@terracoda - Thoughts:
- I like your latest ordering (with sim specific features on top, horizontal line, the PhET link).
- I think "Report Problem" sounds fine to me. I also like you're approach of having helpful text associated with the buttons that likely need more context.
- I love the inclusion of icons also. For the "snapshot," "fullscreen," and "report problem" I can go ahead and create an issue for one of our graphic designers to create these icons - since I think they're pretty straightforward. I like that the features with "..." indicate that you're about to get more information, and for those without "..." the visual icon provides a related cue.

**Comment by emily-phet** [comment]
Regarding some of @jhung comments:

> Some dialogs are inconsistent in the use of the X close button. Options has it, the others do not.

@jessegreenberg - can all the dialogs that we're working on for the PhET menu have an "X" to close? I think this is better for usability, but also the PhET version of the X to close is playful, and keeping the dialogs friendly can be help keep little kiddos who happen upon these options from feeling like they've done something wrong. (the classic windows error message look can freak out little kids)

> Not clear what Projector Mode actually does. 

I'd like to have a sentence with some clarifying information in smaller text under this option. Needs to concisely and in a friendly way convey:
- this option inverts the colors
- can increase visibility for projection
- (if possible) also can be useful for printing sim screenshots
How about this as a start?
"Selecting this option inverts the sim's colors. This can be useful for increasing visibility when projecting the sim, and for taking sim screenshots for handouts that will be printed."

> In the Molecule Shapes sim, there's an Option panel on the sim itself, as well as in the menu. This can cause confusion.

Good catch! I haven't heard of this being a problem (e.g., no issues to phethelp mentioning this). I think the options on the screen are "screen specific" options, while in the phet menu are "sim specific" options. Not sure if this is clear to everyone, though. From my experience with students, they don't really attend to the (visual) headings of the menus, and when we create descriptions for sims with menus labeled as "options", we can create a more specific label.

**Comment by terracoda** [comment]
@emily-phet, glad you like it. That would be great if someone else could do the actual artwork. 

I was not sure if the PhET Website link needs an icon. I was thinking that the little paper air plane would do if the designer thinks an icon is necessary. I left off the _dot dot dots_ because all the other ones like that go to a modal dialog in the same app. The PhET link goes somewhere else entirely.

**Comment by emily-phet** [comment]
I made an issue for artwork in phetsims/tasks/issue#752 (not sure how to reference issues in other repos).

**Comment by terracoda** [comment]
@emily-phet, I think just using the full URL will work: 
[https://github.com/phetsims/tasks/issues/752](https://github.com/phetsims/tasks/issues/752)

**Comment by terracoda** [comment]
> "Selecting this option inverts the sim's colors. This can be useful for increasing visibility when projecting the sim, and for taking sim screenshots for handouts that will be printed."

Suggest shortening to:
"Inverts sim colors to dark on light to improve visibility when using a projector, or to save ink when making screenshots for printed handouts."

The label would be read out as something like:
Project Mode, checkbox not checked.

**Comment by jessegreenberg** [comment]
> "Inverts sim colors to dark

@terracoda @emily-phet this isn't technically true, black might switch to white, but it also might just make some other colors brighter or a different color entirely.

> can all the dialogs that we're working on for the PhET menu have an "X" to close

Yes, they can have a close button. Here is the design document which shows what it will look like.

https://docs.google.com/document/d/19h0U5iqeChirUvVRaudUbmSP_J3yMn7FgnILWrUq9-c/edit

**Comment by terracoda** [comment]
@jessegreenberg, looks like a thorough discussion is taking place. Mention of the Escape key as a close action would be a good addition, and that the alt text for the close button should be "Close" and not "X".

@emily-phet and @jessegreenberg... As for descriptive text for projector mode, maybe keeping it less specific is preferable then. How about:
> Inverts sim colors to improve visibility when using a projector. Projector Mode may also be helpful when making screenshots for printed handouts.

**Comment by emily-phet** [comment]
@terracoda For the projector mode help text, how about:

"Changes sim colors to improve viewing when using a projector. Projector Mode may also be helpful when printing screenshots of the sim."

This makes it a little less specific about how the colors are changed, but I think reads a bit friendlier. What do you think?

**Comment by terracoda** [comment]
@emily-phet, nice. I think the language simpler language as well. Very nice.

**Comment by terracoda** [comment]
@emily-phet, @jessegreenberg, can I close this issue?

Summary to relabel items in the PhET Menu in this issue:

- We have new label text in https://github.com/phetsims/a11y-research/issues/12#issuecomment-265813065, and design work is happening in phetsims/tasks#752
- We clarified description for projector mode in https://github.com/phetsims/a11y-research/issues/12#issuecomment-266741288
- We all agree that work on making the dialogs consistent (https://github.com/phetsims/a11y-research/issues/12#issuecomment-265838527) is important and is ongoing elsewhere

**Comment by emily-phet** [comment]
@terracoda Sounds good to me. I'll go ahead and close.

**Comment by jessegreenberg** [comment]
Reopening after some additional discussion on 12/20/16:

 - 'Check for updates' is confusing, it sounds like it will actually update the sim.  We should consider different labels for this one.
 -

**Comment by terracoda** [comment]
In summary, we have the following for the content in the Nav Bar and PhET Menu
- NAME of SIM
- (Tabs for Screens - not fully discussed yet)
- Hot Keys and Help: Tips on how to use this sim with a keyboard.
- PhET Menu: Tools and links for this sim.

Then the actual menu
- View Options...
- Check Sim Version...
- Take Screenshot (with camera icon)
- Full Screen (with full screen icon)
- Report Problem (with envelope icon)
- About this Sim...
Visual separation here
- PhET Website 

Will add updated pencil sketch in a second.

**Comment by terracoda** [comment]
@jessegreenberg, I think I got everything in the last comment. Reassigning to you.

**Comment by emily-phet** [comment]
@jessegreenberg @terracoda After checking with @ariel-phet it seems like "check for updates" actually does check for updates, and allows the user to update the sim (e.g., presumably for something downloaded for offline use). I've asked @ariel-phet his thoughts on moving this option into the "about this sim" dialog, I'll update this once I hear back.

**Comment by terracoda** [comment]
@jessegreenberg, I think already is in the "About the sim" dialog :-) If it actually checks for updates, we can just leave the label as is.

**Comment by jessegreenberg** [comment]
Yes, the "Check for updates" button does check for updates.  If the sim is out of date, the user will see something like

![capture2](https://cloud.githubusercontent.com/assets/6396244/21872229/8b35401a-d824-11e6-9d41-5a6a8ca58a67.PNG)

The "Get Update" button behaves the same as the "New version available" link in the dialog.
![capture2](https://cloud.githubusercontent.com/assets/6396244/21872377/6eaf3ecc-d825-11e6-8114-9e39a43318a8.PNG)

@terracoda would you mind describing why each of these labels need to be changed? For instance "Report Problem" doesn't seem too different from "Report a Problem".  The change will require translators to revisit each of these strings, so it would be best to keep the same labels where possible.

**Comment by terracoda** [comment]
@jessegreenberg, the removal of indefinite and definite articles ("a", "an" and "the" ) is a recommendation I took from the [Diagram Centre's report](http://diagramcenter.org/accessible-dynamic-scientific-graphics.html) by  Kyle Keane. In verbalization it reduces the number of syllables spoken without loss of meaning. This is something I tried to be consistent about while designing labels and phrases for _Balloons and Static Electricity_.
Quotes from the report:
> Concise—use phrases that are as simple as possible. Remove any extraneous transitions or fluff. (page 13)
> No superfluous words are used to construct full sentences (page 16).

Note that, this may already be happening naturally for many labels and phrases in translation. I am unsure of the full effect this strategy may have on translation. Some languages do not even have definite and indefinite articles.

I think consistency is the best policy.

This strategy may be beneficial for visual representations, as visual space is often limited. "Report a problem" and "Take a screenshot" might wrap to two lines if the "a" is included, for example.

Is it possible to review sims, one a time as descriptions are designed? Of course, this particular menu is for all sims. Maybe @emily-phet would like to way in on this?

I personally prefer shorter labels and phrases wherever possible, but still have limited experience with the number of sims I have seen.

**Comment by terracoda** [comment]
@jessegreenberg, the image of the form you shared is nice and clear. It, too, will need to be converted to text.

We can do it straight as it is with the bold text as the heading, or we can go more structural with a general heading. Will post something this evening (my time).

**Comment by emily-phet** [comment]
@jessegreenberg - thanks for reminding about the translation that needs to occur. Since we definitely want to change at least one of the labels in the PhET Menu (I think "About" to "About this Sim" is significant), I think we should be 100% certain of all the changes we want and then do them all at once - even if they seem minor. 
It doesn't seem like I updated this thread with @ariel-phet response about moving "Check for Updates" into the "About this Sim" dialog. He thought this would take a significant amount of infrastructure effort, and did not want to do this. If that's the case, we should leave it, but I still think it would be a lot cleaner in the "About this Sim" dialog than in the menu. 
@terracoda - what do you think about moving "Check for Updates" to just above "About this Sim".

**Comment by terracoda** [comment]
@emily-phet, @jessegreenberg, the sim version is stated in the _About this Sim_ dialog (as mentioned in https://github.com/phetsims/a11y-research/issues/12#issuecomment-265838527), so it makes perfect sense to move _Check Sim Version_  (or _Check for Updates_) to be next to _About this Sim_. It could even go below it with some indentation, unless there is a reason for not nesting the menu items.

<img width="589" alt="about-dialog-jt" src="https://cloud.githubusercontent.com/assets/1902116/21915905/cadcb27c-d962-11e6-84f2-ecb38111ff25.png">

And since, the _Check for Updates_ dialog **does actually** check for updates (https://github.com/phetsims/a11y-research/issues/12#issuecomment-272042307), would the item's placement in the menu be enough to clarify its function addressing @jhung's concern in https://github.com/phetsims/a11y-research/issues/12#issuecomment-265838527, and thus also minimizing the impact on translation? 

See four possible versions below (changed items italicized).

**V1: Change label and change in position to above About**
- View Options...
- Take Screenshot (with camera icon)
- Full Screen (with full screen icon)
- Report Problem (with envelope icon)
- **_Check Sim Version..._** 
- About this Sim...

Visual separation here
- PhET Website

**V2: Change in label and change in position to below About**
- View Options...
- Take Screenshot (with camera icon)
- Full Screen (with full screen icon)
- Report Problem (with envelope icon)
- About this Sim...
  - _**Check Sim Version...**_ 

Visual separation here
- PhET Website

**V3: NO change in label, and change in position to above About**
- View Options...
- Take Screenshot (with camera icon)
- Full Screen (with full screen icon)
- Report Problem (with envelope icon)
- **_Check for Updates..._** 
- About this Sim...

Visual separation here
- PhET Website

**V4: NO change in label, and change in position to below About **
- View Options...
- Take Screenshot (with camera icon)
- Full Screen (with full screen icon)
- Report Problem (with envelope icon)
- About this Sim...
  - **_Check for Updates..._**

Visual separation here
- PhET Website

**Comment by emily-phet** [comment]
@terracoda Thanks for the suggestions! While I agree that moving the "Check for Updates" into the "About this Sim" dialog, @ariel-phet thought this would require infrastructure changes (maybe in Joist?) that we don't have the resources for at the moment. Does this sound right to you @jessegreenberg? 
Regarding label - I think that "Check for Updates" is the most clear, since that's the most impactful action you will likely want to when selecting this option. You will also get some info about the version, but that's less significant than actually updating and changing the version. 
I'm not sure how I feel about the visual indentation of the "Check for Updates" option below the "About this Sim" option. Conceptually it makes sense, but it seems a bit unusual to have an indented option in a list that pops up like this, and it might require that the whole menu take up more horizontal space since "Check for Updates..." is kind of long in comparison to the other options. @jessegreenberg can you weigh in on this indented list idea?

**Comment by jessegreenberg** [comment]
> moving the "Check for Updates" into the "About this Sim" dialog...would require infrastructure changes

Not too much, it is pretty straight forward with the current implementation.  But this would take more work than just reordering, which would be trivial.  We would need to decide what the new button would look like and how it would replace the link that is in the dialog now.

But there is another advantage to having the "Check for Updates..." button in the menu, which is that it is very easy to find.  When the sim is out of date the button turns green with the hamburger/three dot icon, making the reasoning behind the bright button color obvious:
![capture](https://cloud.githubusercontent.com/assets/6396244/21946540/1c648d14-d99e-11e6-8720-dffc848e5357.PNG)

>  Conceptually it makes sense, but it seems a bit unusual to have an indented option in a list that pops up like this,

I agree, the desired effect makes sense, but it seems unusual to have an indented button in this list.

**Comment by jessegreenberg** [comment]
> the removal of indefinite and definite articles ("a", "an" and "the" ) is a recommendation I took from the Diagram Centre's report by Kyle Keane.

Just checking other menu systems, it looks like Firefox takes this approach while Chrome is more verbose with "a", "an" and "the".

**Comment by jessegreenberg** [comment]
@terracoda I like V3 in https://github.com/phetsims/a11y-research/issues/12#issuecomment-272348566!

**Comment by terracoda** [comment]
@jessegreenberg, @emily-phet, I wasn't suggesting we move _Check for Updates_ into the _About this Sim_ dialog. The versioning information is already stated there, so moving the item next to the _About this Sim_ item makes good sense. Indentation makes a parent-child relationship between the items more clear, but that seems unnecessary here! The menu is a handy list of tools, and a place to get more information about the sim.

@jessegreenberg, thanks for the added information about how the menu item works. I hadn't seen the green highlighting before. I was actually wondering if the menu item should be more dynamic to let the users know that an update is available. As you have demonstrated, it already does this, but it uses color alone to communicate the message.

Has the design team considered coupling the color change with the warning icon? This would be more compliant with WCAG's success criteria [1.4.1 Use of Color ](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#content-structure-separation).

The icon would require a description (or alt text of some kind) for the non-visual environment which would affect translation, but would not immediately affect any translation in the visual mode. @emily-phet and @jessegreenberg, what do you think of this idea?

And for the record, I also prefer version 3 which is what @emily-phet originally suggested. I like having the _About this Sim_ item last. I think we are all in agreement for the ordering and wording.

**Comment by terracoda** [comment]
@jessegreenberg, thanks for checking different browsers on their use of definite and indefinite articles. 

There are many instances where definite and indefinite articles are required for clear meaning, but menu items and headlines often do fine without them [language change](http://english.stackexchange.com/questions/116750/have-there-been-any-movements-tendencies-to-remove-definite-and-indefinite-artic).

This change in the English version of PhET Sims does not necessarily have an effect on translated versions. Individual languages handle this in their own way.

I think we should be consistent, so if there is agreement to remove them, we should be consistent about their removal, and check to see where this may impact translation.

**Comment by emily-phet** [comment]
@terracoda @jessegreenberg 
Just wanting to summarize a bit:
- Wording to be included in the menu:
View Options...
Take Screenshot (with camera icon)
Full Screen (with full screen icon)
Report Problem (with envelope icon)
Check for Updates...
About this Sim...

Everything left aligned, no indentation. (note to @terracoda - my understanding is that when changes are made, a notification is sent to our group of trusted translators to address these changes...so even if no change is needed in the translation the volunteers would need to take a look. I don't think this is a problem as long as we only do this once - for the foreseeable future - for the PhET menu.)

- Color/images - consider warning icon along with color change 
I didn't realize this changed color. We might consider an icon when the sim is up to date, and another when the sim is not. We definitely should *not* have only a color change as an indicator.

Are there any other remaining questions in this issue?

**Comment by terracoda** [comment]
@emily-phet, I think your summary is good.

I think it is worth considering 2 icons for _Check for Updates_. Two icons would have a bigger impact on translation, and verbosity levels, as alt text would be needed for the icons. Two possible examples below:

**Example 1**
- Check for Updates... (no icon)
- Check for Updates... (warning icon with alt text: "Update available.")

**Example 2**
- Check for Updates... (icon with alt text like "Already up to date.")
- Check for Updates... (icon with alt text: "Update available.")

**Comment by terracoda** [comment]
I think Example 1 keeps things simpler. If we consider two icons, we might also just want to use the label to describe the change directly, rather than using alt text.

For example:
- Sim is Up to Date... (icon with no alt text)
- Sim Updates Available... (warn icon with no alt text)

I am not suggesting we do this, but this approach is less verbose for non-visual users.

**Comment by emily-phet** [comment]
@terracoda  if we can (relatively quickly) come up with two icons that clearly indicate "up to date" and "not up to date", and it's relatively straightforward to implement a two-icon option for @jessegreenberg , I think we should go with this option.

Going forward, our goal for work on these common areas of the sim is to ensure that these are accessible, and to try to make all the changes necessary for the foreseeable future at once (rather than leaving issues that will need to be addressed later). So no need to propose minimally accessible options. Also, if we do this well once, it's fine to have to update the translations, so since we're making at least one change that will require translation, we might as well not shy away from others that will improve the accessibility. So, no need to worry about minimizing translations at this point - we're already going to have to request translation updates so whether it's for two phrases or 10 isn't a huge difference.

**Comment by jhung** [comment]
What do you all think of "Check version" vs. "Check for updates"?

I like the idea of having an indicator much like that seen on smart phones. A simple circle with an up arrow may be enough? Or maybe an "!"? Like:

[http://plainicon.com/download-icons/54752/plainicon.com-54752-b2fe-64px.png](url)
[http://www.freeiconspng.com/uploads/alert-icon--icon-search-engine-20.png](url)

**Comment by terracoda** [comment]
@emily-phet, for consistency I think we can stick with the adapted Font-awesome icons that the simulations are already using, the green checkbox with a white check and the orange triangle with an exclamation which are based on the following 2 icons:
- http://fontawesome.io/icon/check-square/
- http://fontawesome.io/icon/exclamation-triangle/

@jhung, the links you posted go to 404 pages for me, so I can't see the icons you are suggesting. Perhaps they are similar to the Font-awesome icons?

Also, @jhung, could you explain why you think the label _Check for Updates_ needs to be changed? Changing the language on the outside menu item would require similar changes on the labels and text for the button and link text in the dialog as well. "Update" is commonly used to indicate a minor version change.

**Comment by terracoda** [comment]
@jhung, actually, the use of the word "version" comes first in both the dialogs (the Update dialog and the About the Sim dialog), so I agree using _Check Version_ makes good sense. I wasn't fond of "Check Sim Version". "Check Version" is simpler and avoids repetition in the alt text for the icons.

**How's this?**
- Check Version (check square icon with alt text "Already up to date.")
- Check Version (exclamation triangle icon with alt text: "Update available.")

**Comment by jhung** [comment]
@terracoda The icons I were thinking of was a circle with an up arrow (indicates an update available), or a circle with an "!" (the triangle seems more like a warning, whereas a circle is more like a note).

@terracoda Re: "Check Version + icon" looks / sounds good to me! But I went and checked with people to see what they thought of "Check for Update" and you can see the responses here:

https://botbot.me/freenode/fluid-design/2017-01-17/?msg=79502839&page=1

Seems like "Check for Updates" may be okay as is.

**Comment by terracoda** [comment]
@emily-phet, I think based on yesterday (March 12) A11y meeting and the comments above that this is the final version of the content for the bottom bar and PhET Menu:

  

- Keyboard Shortcuts: Tips on how to use this sim with a keyboard.
- PhET Menu: Tools and links for this sim.

Then the actual menu items in the the PhET Menu

-  View Options...
-  Check for Updates (check square icon with alt text "Already up to date.")
-  Take Screenshot (camera icon)
-  Full Screen (full screen icon)
-  Report Problem (email icon)
-  About this Sim...

Visual separation here
- PhET Website

**Note** there are 2 icons for Check for Updates
- Check for Updates  (exclamation triangle icon with alt text: "Update available.")

@jessegreenberg and @emily-phet, is this summary correct?

**Comment by emily-phet** [comment]
@terracoda I sent an email yesterday to Ashley to get her feedback on "Keyboard Shortcuts" vs "Key Commands" or something else. Let's see if she responds some time today. I'll update here if/when I hear back.

**Comment by terracoda** [comment]
@emily-phet, OK, good. One thing I did not do until just now was actually look up the words (silly I didn't do this before the meeting). While there can be subtle differences depending in what platform one is using, the terms Hot Keys and Keyboard Shortcuts are pretty interchangeable. 

I was thinking narrowly about the meaning of Hot Keys and that they were specifically only custom key presses kike the ones in Balloons and Static Electricity.

Let's see what Ashley has to say. What we I think we all agreed upon was that we did not need the word "Help."

[Keyboard Shortcut on Wiktionary](https://en.wiktionary.org/wiki/keyboard_shortcut) 
[Keyboard Shortcuts on Wikipedia](https://en.wikipedia.org/wiki/Keyboard_shortcut)
